### ISSUE TITLE
Layout test can contain whitespace inside the function name which causes unexpected token error

### ISSUE THREAD

**Issue by upsuper** [servo/rust-bindgen#844]
### Input C/C++ Header

```C++
template<typename T>
class Foo {
  T* m;
};

class Bar {
  Foo<const char> m;
};
```

### Bindgen Invocation

```
$ bindgen input.hpp
```

### Actual Results

```rust
#[repr(C)]
#[derive(Debug, Copy, Clone)]
pub struct Foo<T> {
    pub m: *mut T,
    pub _phantom_0: ::std::marker::PhantomData<::std::cell::UnsafeCell<T>>,
}
#[repr(C)]
#[derive(Debug, Copy)]
pub struct Bar {
    pub m: Foo<::std::os::raw::c_char>,
}
#[test]
fn bindgen_test_layout_Bar() {
    assert_eq!(::std::mem::size_of::<Bar>() , 8usize , concat ! (
               "Size of: " , stringify ! ( Bar ) ));
    assert_eq! (::std::mem::align_of::<Bar>() , 8usize , concat ! (
                "Alignment of " , stringify ! ( Bar ) ));
    assert_eq! (unsafe {
                & ( * ( 0 as * const Bar ) ) . m as * const _ as usize } ,
                0usize , concat ! (
                "Alignment of field: " , stringify ! ( Bar ) , "::" ,
                stringify ! ( m ) ));
}
impl Clone for Bar {
    fn clone(&self) -> Self { *self }
}
#[test]
fn __bindgen_test_layout_Foo_open0_const char_close0_instantiation() {
    assert_eq!(::std::mem::size_of::<Foo<::std::os::raw::c_char>>() , 8usize ,
               concat ! (
               "Size of template specialization: " , stringify ! (
               Foo<::std::os::raw::c_char> ) ));
    assert_eq!(::std::mem::align_of::<Foo<::std::os::raw::c_char>>() , 8usize
               , concat ! (
               "Alignment of template specialization: " , stringify ! (
               Foo<::std::os::raw::c_char> ) ));
}
```

The last layout test function contains a whitespace in its name.

### Expected Results

It should generate valid function name.

**Comment by upsuper** [comment]
This is likely a regression from #829. cc @Manishearth

**Comment by upsuper** [comment]
Pointed by @emilio that this is fixed by a commit in #841.
